app/parse.py

```diff
@@ -8,5 +13,46 @@
     duration: str
 
 
-def get_all_courses() -> list[Course]:
-    pass
+def parse_courses(course_elements: Tag) -> Course:
+    name = course_elements.select_one(".typography_landingH3__vTjok").text
+    short_description = course_elements.select_one(
+        ".typography_landingTextMain__Rc8BD.mb-32"
+    ).text.strip()
+    duration_elements = course_elements.select(
+        ".typography_landingTextMain__Rc8BD."
+        "ProfessionCardTags_regularTag__yTc6K"
+    )
+
+    if len(duration_elements) > 1:
+        duration = duration_elements[1].text.strip()
+    else:
+        duration = (duration_elements[0].
+                    text.strip()) if (
+            duration_elements) else "Duration not found"
```

Comment:
The logic here assumes that `duration_elements` will always have at least one element. If `duration_elements` is empty, this will raise an `IndexError`. Consider adding a check to ensure `duration_elements` is not empty before accessing its elements.